engine/lib/entities.php

```diff
@@ -271,43 +271,19 @@ function update_subtype($type, $subtype, $class = '') {
  * @param string $subtype        The subtype of the entity we want to create (default: 'all')
  *
  * @return bool
+ * @deprecated 2.2
  */
 function can_write_to_container($user_guid = 0, $container_guid = 0, $type = 'all', $subtype = 'all') {
-	$container_guid = (int)$container_guid;
+	elgg_deprecated_notice(__FUNCTION__ . ' is deprecated. Use ElggEntity::canWriteToContainer()', '2.2');
 	if (!$container_guid) {
 		$container_guid = elgg_get_page_owner_guid();
 	}
-
 	$container = get_entity($container_guid);
-
-	try {
-		$user = _elgg_services()->entityTable->getUserForPermissionsCheck($user_guid);
-	} catch (UserFetchFailureException $e) {
+	if (!$container) {
```

Comment:
I am not sure how this hasn't xaused problems before.  No page owner would pass empty container to the hook.